pkg/bff/users.go

```diff
@@ -53,121 +68,213 @@ func (s *Server) Login(c *gin.Context) {
 	appdata := &auth.AppMetadata{}
 	if err = appdata.Load(user.AppMetadata); err != nil {
 		log.Error().Err(err).Msg("could not parse user app metadata")
-		c.JSON(http.StatusInternalServerError, "could not parse user app metadata")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not parse user app metadata"))
+		return
+	}
+
+	// Fetch the current user role.
+	if roles, err = s.auth0.User.Roles(*user.ID); err != nil {
```

Comment:
I'm going to need a little more time with this logic; so I'm going to stop the review here so you can see and resolve the other comments. I'll try to get to this part of the review soon!